FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Core/Jobs/AzureTableMetadataStore.cs

```diff
@@ -134,6 +135,32 @@ public async Task<CompartmentInfoEntity> GetCompartmentInfoEntityAsync(byte queu
             return (await _metadataTableClient.GetEntityAsync<CompartmentInfoEntity>(TableKeyProvider.CompartmentPartitionKey(queueType), TableKeyProvider.CompartmentRowKey(patientId), cancellationToken: cancellationToken)).Value;
         }
 
+        public async Task<JobStatusEntity> GetJobStatusAsync(byte queueType, long groupId, long jobId, CancellationToken cancellationToken = default)
+        {
+            JobStatusEntity entity = null;
+            try
+            {
+                Response<JobStatusEntity> response = await _metadataTableClient.GetEntityAsync<JobStatusEntity>(
+                    TableKeyProvider.JobStatusPartitionKey(queueType, groupId),
+                    TableKeyProvider.JobStatusRowKey(groupId, jobId),
+                    cancellationToken: cancellationToken);
+
+                entity = response.Value;
+            }
+            catch (RequestFailedException ex) when (ex.ErrorCode == AzureStorageErrorCode.GetEntityNotFoundErrorCode)
+            {
+                _logger.LogInformation(ex, "The job status entity doesn't exist, will create a new one.");
+            }
+            catch (RequestFailedException ex)
+            {
+                _logger.LogError(ex, $"Get job status entity failed. Reason : {ex.Message}");
+                throw new SynapsePipelineInternalException($"Get job status entity failed. Reason : {ex.Message}", ex);
```

Comment:
we define a [`MetadataStoreException`](https://github.com/microsoft/FHIR-Analytics-Pipelines/blob/main/FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Core/Exceptions/MetadataStoreException.cs), would it be more suitable to throw a MetadataStoreException here?

Reply:
Updated here, thanks.